lib/redis_memo/memoize_query/cached_select.rb

```diff
@@ -167,20 +167,20 @@ def map_substitutes(values)
   # @param sql [String] SQL query
   # @return [Boolean] indicating whether a query should be cached
   def self.extract_bind_params(sql)
-    ast = RedisMemo::ThreadLocalVar.arel&.ast
-    return false unless ast.is_a?(Arel::Nodes::SelectStatement)
-    return false unless ast.to_sql == sql
-
-    RedisMemo::ThreadLocalVar.substitues ||= {}
-    # Iterate through the Arel AST in a Depth First Search
-    bind_params = extract_bind_params_recurse(ast)
-    return false unless bind_params
-
-    bind_params.uniq!
-    return false unless bind_params.memoizable?
-
-    RedisMemo::ThreadLocalVar.arel_bind_params = bind_params
-    true
+    RedisMemo::Tracer.trace('redis_memo.memoize_query.extract_bind_params', nil) do
+      ast = RedisMemo::ThreadLocalVar.arel&.ast
+      return false unless ast.is_a?(Arel::Nodes::SelectStatement)
+      return false unless ast.to_sql == sql
+
+      RedisMemo::ThreadLocalVar.substitues ||= {}
+      # Iterate through the Arel AST in a Depth First Search
+      bind_params = extract_bind_params_recurse(ast)
```

Comment:
How about renaming `extract_bind_params_recurse`? Maybe to  `construct_bind_params_recurse` or something similar, but since we don't actually extract the params until `extract!` is called

Reply:
Good point!